iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepLexicalGrammar.java

```diff
@@ -98,15 +110,19 @@ private static void lexical(LexerlessGrammarBuilder b) {
         b.skippedTrivia(b.regexp("[" + LexicalConstant.LINE_TERMINATOR + LexicalConstant.WHITESPACE + "]*+"))))
       .skip();
 
+    b.rule(IDENTIFIER_LITERAL).is(SPACING, b.regexp(BicepLexicalConstant.IDENTIFIER_LITERAL));
+    b.rule(QUOTED_STRING_LITERAL).is(SPACING, b.regexp(BicepLexicalConstant.QUOTED_STRING_LITERAL_NO_QUOTES));
     b.rule(ALPHA_NUMERAL_STRING).is(SPACING, b.regexp(BicepLexicalConstant.ALPHA_NUMERAL_STRING));
-    b.rule(STRING_LITERAL_VALUE).is(b.regexp(BicepLexicalConstant.STRING));
-    b.rule(NUMERIC_LITERAL_VALUE).is(b.regexp(BicepLexicalConstant.NUMBER));
-    b.rule(TRUE_LITERAL_VALUE).is(b.regexp(BicepLexicalConstant.TRUE));
-    b.rule(FALSE_LITERAL_VALUE).is(b.regexp(BicepLexicalConstant.FALSE));
-    b.rule(NULL_LITERAL_VALUE).is(b.regexp(BicepLexicalConstant.NULL));
+    b.rule(STRING_LITERAL_VALUE).is(SPACING, b.regexp(BicepLexicalConstant.STRING));
+    b.rule(NUMERIC_LITERAL_VALUE).is(SPACING, b.regexp(BicepLexicalConstant.NUMBER));
+    b.rule(TRUE_LITERAL_VALUE).is(SPACING, b.regexp(BicepLexicalConstant.TRUE));
+    b.rule(FALSE_LITERAL_VALUE).is(SPACING, b.regexp(BicepLexicalConstant.FALSE));
+    b.rule(NULL_LITERAL_VALUE).is(SPACING, b.regexp(BicepLexicalConstant.NULL));
+
+    b.rule(OPENING_APOSTROPHE).is(SPACING, Punctuator.APOSTROPHE);
```

Comment:
This rule is unnecessary because you already prefixed APOSTROPHE with a SPACE in line 99.